src/problems/sudoku_Leuni.clj

```diff
@@ -0,0 +1,234 @@
+;              in the current-pos ([row, col]) row value.
+;
+; Returns    : true  - if no duplicates (all elements are distinct)
+;              false - if duplicates (all elements are not distinct)
+(defn check-row[state pos]
+  (let [board (:board state)
+        cur-pos pos]
+
+    ; Get the current row index from the current-pos
+    (def row-index (first cur-pos))
+
+    ; Get the current row collection by using the row index
+    (def row (get board row-index))
+
+    ; Determine if there are no duplicate values
+    ; that are greater than zero
+    (apply distinct? (filter #(> % 0) row))
+
+  ))
+
+
+; Function   : check-column
+;
+; Description: Given a state [board current-pos]
+;              function will determine if there are any duplicate values
+;              in the current-pos ([row, col]) column value.
+;
+; Returns    : true  - if no duplicates (all elements are distinct)
+;              false - if duplicates (all elements are not distinct)
+(defn check-column [state pos]
+  (let [board (:board state)
+        cur-pos pos]
+
+    ; Pull the values from the current column
+    ; and store them in a vector
+    (def column (map #(nth % (second cur-pos)) board))
+
+    ; Determine if there are no duplicate values
+    ; that are greater than zero
+    (apply distinct? (filter #(> % 0) column))
+
+  ))
+
+; Function   : isLegal?
+;
+; Description: Given a state [board current-pos] and old-pos
+;              tests the logical And of the three tests
+;              check-row, check-column and check-square.
+;
+; Returns    : true  - if no duplicates (all elements are distinct in rows,columns and squares)
+;              false - if duplicates (all elements are not distinct within rows columns and squares)
+(defn isLegal? [state old-pos]
+  (and (check-row state old-pos) (check-column state old-pos) (check-square state old-pos)))
+
+; Function   : children
+;
+; Description: Given a state [board current-pos]
+;              creates all possible children and
+;              then filters them based on their legality
+;
+; Returns    : All legal children
+(defn children [state]
+  (let [old-pos (:current-pos state)]
+    (filter #(isLegal? % old-pos) (all-pos state))
+        )
+  )
+
+; Function    : get-previous-position
+;
+; Description : Takes the current-pos from a state and returns the previous position
+(defn get-previous-position [state]
+  (let [cur-pos (:current-pos state)
+        cur-row (get cur-pos 0)
+        cur-col (get cur-pos 1)]
+
+    (if (and (= cur-row 0) (= cur-col 0)) cur-pos
+      (if (= cur-col 0) [(dec cur-row) 8]
+          [cur-row (dec cur-col)]))
+
+  )
+
+)
+
+; Function    : num-wrong
+;
+; Description : A heuristic for determining how many values are wrong in the current working row.
+;               The program should pursue Sudoku boards with the least number of incorrectly
+;               placed values.
+(defn num-wrong [goal-state current-state]
```

Comment:
This works as a heuristic, but is kind of cheating because it essentially is saying "Hey, copy these answers over to your sheet".